### PR TITLE
NUI-4129: Focus outline styles

### PR DESCRIPTION
**Pull request by NataliiaPodpriatova**

### CODE REVIEW

File: packages/bits/src/lib/menu/menu-item/menu-option/menu-option.component.less [hunk 1]
```diff
@@ -3,26 +3,39 @@
...
+
         .nui-checkbox__translucent {
             cursor: not-allowed;
         }
     }
 
     &.nui-menu-item--disabled {
         > .nui-menu-item__option {
+            padding-left: @nui-space-xs;
+            padding-right: @nui-space-xs;
```
Comment (Vita-Karas) [inline packages/bits/src/lib/menu/menu-item/menu-option/menu-option.component.less:34]: Can be changed to `padding: 0 @nui-space-xs;`

File: packages/bits/src/lib/selector/selector.component.less [hunk 2]
```diff
@@ -16,16 +16,29 @@
...
 
-    &__checkbox-container {
-        margin: -(@nui-button-border-width) 0;
+    .nui-button__content {
+        width: 100%;
     }
+    .nui-checkbox__label {
+            width: 100%;
+            border-top-right-radius: 0;
+            border-bottom-right-radius: 0;
```
Comment (Vita-Karas) [inline packages/bits/src/lib/selector/selector.component.less:33]: Can be simplified

File: packages/bits/src/lib/menu/menu-item/menu-option/menu-option.component.less (hunk 1 above)
Comment (NataliiaPodpriatova) [reply packages/bits/src/lib/menu/menu-item/menu-option/menu-option.component.less:34]: Thanks, will be changed

File: packages/bits/src/lib/selector/selector.component.less [hunk 3]
```diff
@@ -16,16 +16,28 @@
...
+            width: 100%;
+            border-radius: @nui-radius-default 0 0 @nui-radius-default;
+            border: @nui-line-default solid @nui-color-line-default;
+            .nui-checkbox__mark {
+                margin: -(@nui-button-border-width) 0;
+            }
+            &:hover {
+                .setCssVariable(background-color, nui-color-bg-transparent-hover);
+            }
+        }
```
Comment (Blakesalot) [inline packages/bits/src/lib/selector/selector.component.less:40]: indentation is off a bit

**Review by Blakesalot — CHANGES_REQUESTED** [review]

File: packages/bits/src/lib/menu/menu-item/menu-option/menu-option.component.less [hunk 4]
```diff
@@ -3,26 +3,38 @@
...
 .nui-menu .nui-menu-item, .nui-menu-popup-host .nui-menu-item, .nui-menu-popup .nui-menu-item {
     .nui-menu-item__option {
-        padding: 0 @nui-space-xs;
+        padding: 0;
         .nui-checkbox__label {
             font-weight: @nui-font-weight-regular;
             font-size: @nui-font-size-default;
             width: 100%;
-        }
+            padding: @nui-space-xs+@nui-space-xxs @nui-space-sm+@nui-space-xxs;
```
[the remark this answers is not in the document]
Comment (NataliiaPodpriatova) [reply packages/bits/src/lib/menu/menu-item/menu-option/menu-option.component.less:11]: Yes, the default paddings for the checkbox label are 7px. Here I just overwrite the default value for the top and bottom and add an additional 5 px for the right and left side for resizing the checkbox's label to the size of the menu item and avoiding case with 2 focuses simultaneously (on the checkbox's label and on menu item)

File: packages/bits/src/lib/menu/menu-item/menu-option/menu-option.component.less [hunk 5]
```diff
@@ -3,26 +3,38 @@
 
 .nui-menu .nui-menu-item, .nui-menu-popup-host .nui-menu-item, .nui-menu-popup .nui-menu-item {
     .nui-menu-item__option {
-        padding: 0 @nui-space-xs;
+        padding: 0;
```
[the remark this answers is not in the document]
Comment (NataliiaPodpriatova) [reply packages/bits/src/lib/menu/menu-item/menu-option/menu-option.component.less:6]: I described the reason in a previous comment. I'm adding these 5px to the checkbox’s label right and left sides. All sizes are saved.

File: packages/bits/src/lib/menu/menu-item/menu-option/menu-option.component.less [hunk 6]
```diff
@@ -3,26 +3,38 @@
...
+                .setCssVariable(background-color, transparent);
+            }
         }
+
+        .nui-menu-item__checkbox {
+            width: 100%;
+                & > .nui-checkbox {
+                    padding: 0;
+                }
+            }
```
[the remark this answers is not in the document]
Comment (NataliiaPodpriatova) [reply packages/bits/src/lib/menu/menu-item/menu-option/menu-option.component.less:24]: Thank you, I will fix it

File: packages/bits/src/lib/menu/menu-item/menu-option/menu-option.component.less [hunk 7]
```diff
@@ -3,26 +3,38 @@
...
 
     &.nui-menu-item--disabled {
         > .nui-menu-item__option {
+            padding: @nui-space-xs;
             .setCssVariable(color, nui-color-off);
-            &:focus {
-                outline: 0;
+           &:active, &:focus {
+               outline: 0;
+               .no-shadow();
```
[the remark this answers is not in the document]
Comment (NataliiaPodpriatova) [reply packages/bits/src/lib/menu/menu-item/menu-option/menu-option.component.less:37]: Thank you, I will fix it

File: packages/bits/src/lib/selector/selector.component.html [hunk 8]
```diff
@@ -3,7 +3,8 @@
             type="button"
             class="nui-selector__checkbox-button"
             [displayStyle]="'default'"
-            (click)="handleSelectorClick($event)">
+            (click)="handleSelectorClick($event)"
+            tabindex="-1">
```
[the remark this answers is not in the document]
Comment (NataliiaPodpriatova) [reply packages/bits/src/lib/selector/selector.component.html:7]: Thank you for pointing out this. The reason is that the focus outlines for the button and the checkbox (with the new focus outline styles) look pretty much the same. The only difference is that for the button, the box-shadow is outward, and for the checkbox, the box-shadow is inward.
Yes, you are right. I will decrease tabindex for the checkbox (-1), but for the button leave 0.

File: packages/bits/src/lib/selector/selector.component.less [hunk 9]
```diff
@@ -16,16 +16,28 @@
         &.btn {
             min-width: @nui-selector-checkbox-button-width;
             width: @nui-selector-checkbox-button-width;
-            padding: 0 @nui-space-xxs 0 @nui-space-xxs;
+            padding: 0;
```
[the remark this answers is not in the document]
Comment (NataliiaPodpriatova) [reply packages/bits/src/lib/selector/selector.component.less:19]: The reason is the same as in the menu option - resizing purpose and avoiding case with 2 focused elements simultaneously

File: packages/bits/src/lib/selector/selector.component.less (hunk 3 above)
Comment (NataliiaPodpriatova) [reply packages/bits/src/lib/selector/selector.component.less:40]: Thank you, I will fix it

File: packages/bits/src/lib/selector/selector.component.ts [hunk 10]
```diff
@@ -105,6 +105,7 @@ export class SelectorComponent implements OnChanges, AfterViewInit, OnDestroy, I
         this.overlay.clickOutside
             .pipe(takeUntil(this.onDestroy$))
             .subscribe(_ => this.overlay.hide());
+        this.checkbox.checkboxLabel.nativeElement.setAttribute("tabindex", "-1");
```
Comment (Blakesalot) [inline packages/bits/src/lib/selector/selector.component.ts:108]: Ideally, there would be an input on the checkbox component to do this behavior. Do you mind creating a ticket to make this improvement in the future?

**Review by Blakesalot — DISMISSED** [review]

**Review by dmytrofirsov — DISMISSED** [review]

**Review by Vita-Karas — APPROVED** [review]

**Review by Blakesalot — DISMISSED** [review]

**Review by ivankozlovskyi — APPROVED** [review]

**Review by dmytrofirsov — APPROVED** [review]

**Review by Blakesalot — APPROVED** [review]

**Pull request merged**